### PR TITLE
Fix nim 1.4

### PR DESCRIPTION
**Pull request by markspanbroek**
Fixes compilation and tests for Nim 1.4.

* fixes capitalization of chronicles log level (`ERROR` instead of `error`)
* ~~removes unused `NodeId.toBytes` function~~
   ~~(it wasn't seen by Nim 1.2.x, and leads to failing tests when it is seen by Nim 1.4.x)~~
   Fixed in #427
* removes restriction to Nim 1.2.x in nimble file

By the way: the fact that the `NodeId.toBytes` function was not seen by Nim 1.2.x might mean that there's a bug lurking in the [`logDistance`](https://github.com/status-im/nim-eth/blob/fe1bb4c4e75c9d20698e767427effa5fc29bbbc7/eth/p2p/discoveryv5/routing_table.nim#L107) function. In reality it is calculating the distance based on the platform endianness, but the existence of the NodeId.toBytes function seems to suggest that it should have used big endian instead.

### CODE REVIEW

File: eth.nimble [hunk 1]
```diff
@@ -4,7 +4,7 @@ description   = "Ethereum Common library"
 license       = "MIT"
 skipDirs      = @["tests"]
 
-requires "nim >= 1.2.0 & <= 1.2.14",
```
Comment (arnetheduck) [inline eth.nimble:7]: before we change this constraint, we need to enable testing on versions we claim to support, as is being done for each library step by step - the plan is to first support multiple versions in CI, then work towards increasing version support in each library - once that is done, we can freely move between nim versions without the risk of one project breaking things for another - see https://discuss.status.im/t/the-road-to-nim-1-6/2305 for a general overview of the migration / upgrade plan

**Comment by arnetheduck** [comment]
> the fact that the NodeId.toBytes function was not seen by Nim 1.2.x might mean that there's a bug lurking in the logDistance function.

this sounds like something  in dire need of a test case (cc @kdeme) and verification that the removal doesn't break something else

File: eth.nimble (hunk 1 above)
Comment (markspanbroek) [reply eth.nimble:7]: I agree, I'll see whether I can add the support for multiple Nim versions to the CI script.

Comment (arnetheduck) [reply eth.nimble:7]: https://github.com/status-im/nim-libp2p/pull/624 should be a useful model - cc @stefantalpalaru is on top of this

**Review by kdeme — COMMENTED** [review]
> By the way: the fact that the `NodeId.toBytes` function was not seen by Nim 1.2.x might mean that there's a bug lurking in the [`logDistance`](https://github.com/status-im/nim-eth/blob/fe1bb4c4e75c9d20698e767427effa5fc29bbbc7/eth/p2p/discoveryv5/routing_table.nim#L107) function. In reality it is calculating the distance based on the platform endianness, but the existence of the NodeId.toBytes function seems to suggest that it should have used big endian instead.

Oops, I had already hit the issue some time back, but seems I lost track of adding the fix. I've opened a PR which should properly fix this (correct it for BE architecture): https://github.com/status-im/nim-eth/pull/427 , thanks for reporting!

Test cases for this do exist by the way (which I assumed are the ones that failed). It is just not being tested on BE architecture, which is why this remained unnoticed. Some of the test cases were taken from go-ethereum, and I've now also added more of those form there.

**Comment by markspanbroek** [comment]
Great, thanks for the quick fix @kdeme!

**Comment by markspanbroek** [comment]
CI is fixed in #429, which also removes the restriction on Nim version.

**Pull request closed without merging**